### PR TITLE
tests: refactor tests to avoid a single huge generate.py

### PR DESCRIPTION
**Pull request by cyphermox**
## Description

Refactor generate.py tests to split it up per device type; to stop dealing with just a single huge unwieldy file.

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [n/a] New/changed keys in YAML format are documented.
- [n/a] \(Optional\) Closes an open bug in Launchpad.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,7 +35,6 @@ clean:
 	rm -rf test-coverage .coverage
 
 check: default linting
-	tests/generate.py
```
Comment (mwhudson) [inline Makefile]: How are the tests run now? By the nose invocation below?

(PS: can we stop using nose? it's dead upstream afaik)

File: tests/generator/generate.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python3
```
Comment (mwhudson) [inline tests/generator/generate.py]: I don't think this file needs a shebang or is particularly well named. base.py maybe?

File: tests/generator/test_args.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+import subprocess
+
+sys.path.append(os.path.dirname(__file__))
```
Comment (mwhudson) [inline tests/generator/test_args.py]: Ick. Doesn't "from .generate import .." work here?

File: tests/generator/test_args.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python3
```
Comment (mwhudson) [inline tests/generator/test_args.py]: There is a case for making this file executable but as it is now it won't do anything. So i'd rather you removed the shebang (running just these tests can be done with the runner)

File: Makefile (hunk 1 above)
Comment (cyphermox) [reply Makefile]: Yup, using nose below. I wasn't aware it was dead, anything to discover the test suite will do. I'll adress that.

File: tests/generator/generate.py (hunk 2 above)
Comment (cyphermox) [reply tests/generator/generate.py]: Copy-pasta fallout. Thanks!

**Pull request merged**